opentech/apply/projects/views.py

```diff
@@ -1,18 +1,48 @@
 from copy import copy
 
+from django.utils import timezone
 from django.utils.decorators import method_decorator
 from django.views.generic import DetailView, UpdateView
 
 from opentech.apply.activity.messaging import MESSAGES, messenger
+from opentech.apply.activity.models import Activity
 from opentech.apply.activity.views import ActivityContextMixin, CommentFormView
 from opentech.apply.users.decorators import staff_required
 from opentech.apply.utils.views import (DelegateableView, DelegatedViewMixin,
                                         ViewDispatcher)
 
-from .forms import ProjectEditForm, UpdateProjectLeadForm
+from .forms import ProjectEditForm, SetPendingForm, UpdateProjectLeadForm
 from .models import Project
 
 
+@method_decorator(staff_required, name='dispatch')
+class SendForApprovalView(DelegatedViewMixin, UpdateView):
+    context_name = 'approval_form'
+    form_class = SetPendingForm
+    model = Project
+
+    def form_valid(self, form):
+        # lock project
+        response = super().form_valid(form)
+
+        messenger(
+            MESSAGES.SEND_FOR_APPROVAL,
+            request=self.request,
+            user=self.request.user,
+            submission=form.instance.submission,
+            project=form.instance,
+        )
+
+        Activity.comments.create(
```

Comment:
This should be handled through the messaging hub, there is an `ActivityAdapter` there that needs updating.